### PR TITLE
macdylibbundler: deleted duplicate dylibbundler

### LINKED ISSUE #119385: Two copies of dylibbundler
there are two copies of (mac)dylibbundler in the nixpkgs tree:
`pkgs/development/tools/misc/macdylibbundler` with @alexfmpe as maintainer and `pkgs/tools/misc/dylibbundler` with @nomeata as maintainer. Presumably one of them should remain and the other should be removed. I'll leave it to the maintainers to work that out, though macdylibbundler seems more fleshed out and was committed more recently.

### PR DESCRIPTION
**Pull request by Radvendii**
I deleted the dylibbundler but left an alias to the macdylibbundler package.

###### Motivation for this change
dylibbundler package is just an outdated version of macdylibbundler (see #119385)
fixes https://github.com/NixOS/nixpkgs/issues/119385

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Radvendii** [comment]
@nomeata @alexfmpe The change we discussed. Feel free to make adjustments.

**Comment by alexfmpe** [comment]
LGTM

**Review by nomeata — APPROVED** [review]

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -2340,7 +2340,7 @@ in
     enableSSH = true;
   };
 
-  dylibbundler = callPackage ../tools/misc/dylibbundler { };
+  dylibbundler = macdylibbundler;
```
Comment (dotlambda) [inline pkgs/top-level/all-packages.nix:2343]: This should go into `aliases.nix`.

Comment (Radvendii) [reply pkgs/top-level/all-packages.nix]: good to know. done.

File: pkgs/top-level/aliases.nix [hunk 2]
```diff
@@ -168,6 +168,7 @@ mapAliases ({
   dvb_apps = throw "dvb_apps has been removed."; # added 2020-11-03
   dwarf_fortress = dwarf-fortress; # added 2016-01-23
   dwm-git = throw "dwm-git has been removed from nixpkgs, as it had no updates for 2 years not serving it's purpose."; # added 2021-02-07
+  dylibbundler = macdylibbundler; # 2021-04-24
```
Comment (dotlambda) [inline pkgs/top-level/aliases.nix:171]:
```suggestion
  dylibbundler = macdylibbundler; # added 2021-04-24
```

**Comment by dotlambda** [comment]
Should we have added @alexfmpe as to `macdylibbunder`'s maintainers?

**Comment by Radvendii** [comment]
Not necessary. See https://github.com/NixOS/nixpkgs/issues/119385#issuecomment-819893483. They said they haven't touched it in years.

**Pull request merged**